src/libopensc/pkcs15-piv.c

```diff
@@ -1166,7 +1167,18 @@ static int sc_pkcs15emu_piv_init(sc_pkcs15_card_t *p15card)
 		prkey_obj.flags = prkeys[i].obj_flags;
 		prkey_obj.user_consent = prkeys[i].user_consent; /* only Sign key */
 
-		if (prkeys[i].auth_id)
+		sc_card_ctl(p15card->card, SC_CARDCTL_PIV_YK_PIN_POLICY, &pin_policy);
+		switch (pin_policy) {
+		case 0x02:
+			/* PIN is checked once for the session */
+			prkey_obj.user_consent = 0;
+			break;
+		case 0x03:
+			/* PIN is verified just before operation */
+			prkey_obj.user_consent = 1;
+		}
+		/* PIN is never checked for operations if PIN policy is set to 0x01 */
+		if (pin_policy != 0x01)
 			sc_pkcs15_format_id(prkeys[i].auth_id, &prkey_obj.auth_id);
```

Comment:
I think this will need also some more love. If I keep this part in, the pkcs11 fails to find my key, while it is visible ok in the pkcs15 layer without the `Auth ID` field:
```
Private EC Key [KEY MAN key]
	Object Flags   : [0x01], private
	Usage          : [0x100], derive
	Access Flags   : [0x1D], sensitive, alwaysSensitive, neverExtract, local
	Algo_refs      : 0
	FieldLength    : 256
	Key ref        : 157 (0x9D)
	Native         : yes
	ID             : 03
	MD:guid        : 0x'30333862646435356335303039396362306465303666333530343033343931320000000000000000'
```

I do not know where did it go and was not able to tweak the above condition to show the key on the pkcs11 layer without the Auth ID.

Looking at the `CARD AUTH` key, it does not have the `SC_PKCS15_CO_FLAG_PRIVATE` flag. What I used and worked for me was:

```suggestion
		if (prkeys[i].auth_id && pin_policy != 0x01)
			sc_pkcs15_format_id(prkeys[i].auth_id, &prkey_obj.auth_id);
		else 
			prkey_obj.flags &= ~SC_PKCS15_CO_FLAG_PRIVATE;
```

I did not  try with the card auth key yet if the default policy will be 0x00 or 0x01. If we will be able to override the policy, we will need to be able to set the reverse of this for this key (add auth_id + `SC_PKCS15_CO_FLAG_PRIVATE` for this key).

Reply:
my intention was to create a key that doesn't need verification of the PIN. I thought that leaving auth_id would be all that's needed. I'll have a second look.

Reply:
The Yubikey 5 NFCs tokens I have are version 5.2.6, but the metadata feature requires 5.3.0 which is per key(slot) and not in the Attestation cert, as there is only one. The metadata feature is what I was referring to. 

What looks strange is in the code  `piv_yk_get_metadata`  the `rbuf` is only 100 bytes.  The metadata returned could have an RSA pub key which will not fit. So 

https://docs.yubico.com/yesdk/users-manual/application-piv/apdu/metadata.html
says Le absent, which does not make sense, I assume it should be 0x00 with a large `rbuf`  and get responses will be done to return the data.

 The call to: `sc_card_ctl(p15card->card, SC_CARDCTL_PIV_YK_PIN_POLICY, &pin_policy);` does not have any way to return an error.

Reply:
removing SC_PKCS15_CO_FLAG_PRIVATE sounds reasonable, I committed that. thank you.

Scrolling down to
https://docs.yubico.com/yesdk/users-manual/application-piv/apdu/metadata.html#examples
we can see some examples for the command (generated with opensc-tool!) and it seems that a) the public key may indeed be present and be quite long and b) tags appear in sorted order so that the PIN policy appears early. So that, at elast should work as expected.

I think we can avoid propagating (and managing) some error code from SC_CARDCTL_PIV_YK_PIN_POLICY, because we do have safe defaults.